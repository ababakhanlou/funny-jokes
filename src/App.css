.App {
  text-align: center;
}

.App-navbar {
}

.App-card {
  display: flex;
  justify-content: center;
}

.App-jokeButton {
}
