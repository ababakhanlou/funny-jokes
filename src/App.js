import React from "react";
import { Button, Card, NavBar } from "./components";
import getJoke from "./api/joke";
import "./App.css";

class App extends React.Component {
  constructor() {
    super();
    this.state = { joke: "" };
    this.getNewJoke = this.getNewJoke.bind(this);
  }

  async getNewJoke() {
    const joke = await getJoke();
    this.setState({ joke });
  }

  async componentDidMount() {
    this.getNewJoke();
  }

  render() {
    return (
      <div className="App">
        <div className="App-navbar">
          <NavBar />
        </div>

        <div className="App-card">
          <Card joke={this.state.joke} />
        </div>

        <div className="App-jokeButton">
          <Button name="Like <3" />
          <Button action={this.getNewJoke} name="NEXT>>>>>" />
        </div>
      </div>
    );
  }
}

export default App;
