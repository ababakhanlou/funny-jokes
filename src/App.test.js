import React from "react";
import renderer from "react-test-renderer";
import App from "./App";

describe("App", () => {
  jest.mock("./components", () => ({ NavBar: () => () => <div>NavBar</div> }));
  jest.mock("./pages/JokeFinder", () => () => <div>JokeFinder</div>);
  jest.mock("./pages/SavedJokes", () => () => <div>SavedJokes</div>);

  it("render correctly", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
