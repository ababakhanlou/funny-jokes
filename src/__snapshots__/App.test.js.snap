// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render correctly 1`] = `
<div
  className="App"
>
  <div
    className="App-navbar"
  >
    <div
      className="NavBar"
    >
      <h1>
        Dard JOKE!!!
      </h1>
      <nav>
        <ul>
          <li>
            <a
              href="/"
              onClick={[Function]}
            >
              Home
            </a>
          </li>
          <li>
            <a
              href="/saved-jokes"
              onClick={[Function]}
            >
              Saved Jokes
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
  <div
    className="App-card"
  >
    <div>
      <div>
        <div
          className="Card"
        >
          <p>
            
          </p>
        </div>
      </div>
      <div>
        <button
          onClick={[Function]}
        >
          Like &lt;3
        </button>
        <button
          onClick={[Function]}
        >
          NEXT&gt;&gt;&gt;&gt;&gt;
        </button>
      </div>
    </div>
  </div>
</div>
`;
